frontend/src/metabase/visualizations/components/ChartSettingsWidgetPopover.tsx

```diff
@@ -0,0 +1,73 @@
+import React from "react";
+import TippyPopover from "metabase/components/Popover/TippyPopover";
+import ChartSettingsWidget from "./ChartSettingsWidget";
+
+import { PopoverRoot, PopoverTitle } from "./ChartSettingsWidgetPopover.styled";
+
+interface Widget {
+  id: string;
+  props: Record<string, unknown>;
+}
+
+interface ChartSettingsWidgetPopoverProps {
+  anchor: HTMLElement;
+  handleEndShowWidget: () => void;
+  widgets: [Widget];
+  currentWidget: Widget;
+  extraWidgetProps: [];
+}
+
+export const ChartSettingsWidgetPopover = ({
+  anchor,
+  handleEndShowWidget,
+  widgets,
+  currentWidget,
+  extraWidgetProps,
```

Comment:
Passing all these 3 props makes `<ChartSettingsWidgetPopover />` more coupled with its parent as it needs to understand all the 3 props and how to use them.

You can avoid this by passing something like a single `widget` as a ReactElement instead. This also helps with the prop drilling situation and makes this component simpler.

What do you think?

Reply:
It's certainly doable. One of my objectives was to get some of the logic out of `<ChartSettings />` to simplify it up a bit, but I agree that this isn't really ideal either. I'll see what I can do :)